packages/react-form/src/hooks/field/reducer.ts

```diff
@@ -88,10 +90,19 @@ export function makeFieldReducer<Value>({
       }
 
       case 'updateError': {
-        return {
-          ...state,
-          error: action.payload,
-        };
+        if (Array.isArray(action.payload)) {
+          const [firstError] = action.payload;
+          return {
+            ...state,
+            error: firstError,
+            allErrors: action.payload,
+          };
+        } else {
+          return {
+            ...state,
+            error: action.payload,
```

Comment:
I think you should set the `allErrors` field in both the cases.

```suggestion
            error: action.payload,
            allErrors: [action.payload],
```

Reply:
good call, I have made this change 👍